src/main/kotlin/ar/edu/unahur/obj2/semillasAlViento/Parcela.kt

```diff
@@ -1,45 +1,43 @@
 package ar.edu.unahur.obj2.semillasAlViento
 
 class Parcela(val ancho: Int, val largo: Int, val horasSolPorDia: Int) {
+
   val plantas = mutableListOf<Planta>()
-  var cantidadPlantas = 0
 
   fun superficie() = ancho * largo
+
   fun cantidadMaximaPlantas() =
-    if (ancho > largo) ancho * largo / 5 else ancho * largo / 3 + largo
+
+    if (ancho > largo) this.superficie() / 5 else this.superficie() / 3 + largo
+
+  fun parcelaTieneComplicaciones() =
+          this.plantas.any { it.horasDeSolQueTolera() < this.horasSolPorDia }
 
   fun plantar(planta: Planta) {
-    if (cantidadPlantas == this.cantidadMaximaPlantas()) {
-      println("Ya no hay lugar en esta parcela")
+    if (this.plantas.count() == this.cantidadMaximaPlantas()) {
+      throw Exception("Error inesperado! No se puese superar la cantidad máxima de plantas")
     } else if (horasSolPorDia > planta.horasDeSolQueTolera() + 2) {
-      println("No se puede plantar esto acá, se va a quemar")
+      throw Exception("Error inesperado!La planta no puede soportar la cantidad de horas de sol recibidas")
     } else {
       plantas.add(planta)
-      cantidadPlantas += 1
     }
   }
+
+  fun esSemillera(): Boolean = this.plantas.all { it.daSemillas() == true }
 }
 
-class Agricultora(val parcelas: MutableList<Parcela>) {
-  var ahorrosEnPesos = 20000
 
-  // Suponemos que una parcela vale 5000 pesos
-  fun comprarParcela(parcela: Parcela) {
-    if (ahorrosEnPesos >= 5000) {
-      parcelas.add(parcela)
-      ahorrosEnPesos -= 5000
-    }
-  }
+class Agricultora(val parcelas:  List<Parcela>) {
 
   fun parcelasSemilleras() =
     parcelas.filter {
-      parcela -> parcela.plantas.all {
-        planta -> planta.daSemillas()
+      parcela -> parcela.esSemillera()
       }
-    }
 
   fun plantarEstrategicamente(planta: Planta) {
-    val laElegida = parcelas.maxBy { it.cantidadMaximaPlantas() - it.cantidadPlantas }!!
-    laElegida.plantas.add(planta)
+    val parcela = this.parcelaElegida()
+    parcela?.plantar(planta)
```

Comment:
O mejor, todo junto:
```suggestion
    this.parcelaElegida()?.plantar(planta)
```